City-of-Helsinki/helsinki-paatokset
# UHF-9055: HDBT 6.2.0 and PHPCS DrupalPractice fixes

## What was done

- Fixes warnings found by the DrupalPractice PHPCS standard.
- Fixes situation where Decision node data could not be viewed when Case node was missing
- Fixes compatibility issues with HDBT 6.2.0:
  - Fixes related to content area spacing and widths 
  - Decision dropdown styling
  - Accordions on Policymaker listing, decision and trustee pages
  - Fix tabbed content lists

## How to install
* Make sure your instance is up and running on correct branch. You should test this with the production database, if possible
  * `git checkout UHF-9055-phpcs-fixes`
* Run `make drush-cr`

## How to test

* [x] Check that migrations and queues work:
```
drush ap:agg decisions --dataset=latest --queue --start=2023-12-14T00:00:00 --end=2024-01-01T00:00:00 -v
drush ap:agg meetings --dataset=latest --queue --start=2023-12-01T00:00:00 --end=2024-01-01T00:00:00 -v
drush queue:run ahjo_api_aggregation_queue -v
drush ap:update decisions 2e3736af-4b54-4da1-8d4c-50fabf0a4f6a -v
drush ap:update decisions b4b8e27f-9763-4fd0-810c-96ada6c1c4b7 -v
drush ap:update decisions 24109ee7-fd60-ccd7-8564-7eb562400000 -v
drush ap:update decisions 83bb174a-3470-c4ec-bbf8-756f31100001 -v
drush mim ahjo_decisionmakers:all -v
drush mim ahjo_trustees:all -v
drush ap:update organization u511053004070vh1 -v
drush ap:gm -v
drush ap:ud -v
drush queue:run ahjo_api_retry_queue -v
drush ap:ud --logic=case -v
```
* [x] Check that the decision dropdown looks normal on this page (only one decision) and that the accordion works: https://helsinki-paatokset.docker.so/fi/asia/hel-2023-015009/2e3736af-4b54-4da1-8d4c-50fabf0a4f6a
* [x] Check that the decision dropdown looks normal on this page and that all the accordions (content + voting results) work: https://helsinki-paatokset.docker.so/fi/asia/hel-2016-012511/b4b8e27f-9763-4fd0-810c-96ada6c1c4b7
* [x] Accordions should work on trustee pages: https://helsinki-paatokset.docker.so/fi/paattajat/arhinpa
* [x] The social media sharing buttons should look normal on the previous pages (no extra padding or icons)
* [x] Accordions and the "load more" button (on some office holder lists and the council members list) should work on this page: https://helsinki-paatokset.docker.so/fi/paattajat
* [x] The policymaker search should not have any extra padding
* [x] Check that the layout and content lists work on article pages: https://helsinki-paatokset.docker.so/fi/kuulutukset-ja-ilmoitukset
* [x] Individual article page layout should also work (no extra padding or misaligned margins)
* Tabbed content lists should work, and when viewed on mobile the year lists should have a "show more" dropdown instead of having the list overflow or go to multiple lines
  * [x] https://paatokset.hel.fi/fi/paatoksenteko/kaupunginmuseon-lausunnot
  * [x] https://paatokset.hel.fi/fi/paattajat/kaupunginvaltuusto/asiakirjat
  * [x] https://paatokset.hel.fi/fi/paattajat/kaupunginvaltuusto/aloitteet
  * [x] https://paatokset.hel.fi/fi/paattajat/kaupunginvaltuusto/keskustelupoytakirjat
* Check that decision nodes work without a case node:
  * Open this decision node: https://helsinki-paatokset.docker.so/fi/asia/hel-2023-015009/2e3736af-4b54-4da1-8d4c-50fabf0a4f6a  
  * Open this case node and delete it: https://helsinki-paatokset.docker.so/fi/asia/hel-2023-015009
  * Run `make drush-cr` just in case
  * [x] Reload the decision node. It should work normally even though some data might be missing
